### PR TITLE
feat: [IA-1] Update notification token when it changes

### PR DESCRIPTION
**Pull request by Undermaken**
## Short description
This PR updates the notification token only when it is needed.

### before
- ❌  the notification token update request is used to check if the current session is still valid or not
- ❌  the update of notification token is requested every app start

### now
- ✔️  the update of notification token is requested only when the current token is different from the one saved in the backend
- ✔️  to check if current session is still valid, it is used the dedicated `checkSession` with the usage of the relative `getSession` API

## List of changes proposed in this pull request
- add `registeredToken` as persisted field to holds the token saved in the backend
- add the relative persistedStore migration `17`
- now the `initializeApplicationSaga` checks if the session is still valid using the result given by `checkSession` and then updates the notification token
- the notification token is updated only when the current one is different from the saved one in the backend

## How to test
- use the last commit on master of dev-server (found a 🐛  in `updateInstallations` API)
- to simulate a new token released by notification service (APNS/Firebase) put this code here https://github.com/pagopa/io-app/blob/008874676e26655efbab4d7c8eae167008048df6/ts/boot/configurePushNotification.ts#L33
```typescript
store.dispatch(updateNotificationsInstallationToken("mynotificationToken"));
  // if isDevEnv, disable push notification to avoid crash for missing firebase settings
  if (isDevEnv) {
    return;
  }
```

### CODE REVIEW

**Review by pietro909 — APPROVED** [review]

**Comment by pietro909** [comment]
@Undermaken good to go from my side. Shall we merge?

File: ts/sagas/notifications.ts
Comment (fabriziofff) [inline ts/sagas/notifications.ts:66]: Atm we track only `updateNotificationInstallationFailure`. Do you think could be useful to track also this action, for troubleshooting purpose?
Or maybe could be useful to track also the "not required update because the token is the same". What you think?

Comment (Undermaken) [reply ts/sagas/notifications.ts:68]: this is a good point for analytics
I guess tracking both could be a good choice.
https://github.com/pagopa/io-app/pull/3295/commits/92c6f38ccb397b4f501bbc2ef60e9d40d7e4d841

**Comment by Undermaken** [comment]
> @Undermaken good to go from my side. Shall we merge?

Waiting for @fabriziofff approval

File: ts/sagas/notifications.ts
Comment (fabriziofff) [reply ts/sagas/notifications.ts:68]: Could be useful to track also when the token is registered with success?

Comment (Undermaken) [reply ts/sagas/notifications.ts:68]: `notificationsInstallationTokenRegistered` represents that event: backend saved the notification token sent by the app
It is also tracked in the analytics middleware
Are you talking about to add another specific action?

Comment (fabriziofff) [reply ts/sagas/notifications.ts:68]: Ops, didn't see in analytics!

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**
